app/src/main/java/com/humberto/tasky/agenda/data/task/TaskRepositoryImpl.kt

```diff
@@ -23,14 +29,37 @@ class TaskRepositoryImpl @Inject constructor(
 
     override suspend fun createTask(task: AgendaItem.Task): EmptyResult<DataError> {
         return try {
-            taskDao.upsertTask(task.toTaskEntity())
-            Result.Success(Unit)
+            val result = safeCall {
+                taskDao.upsertTask(task.toTaskEntity())
+                agendaApi.createTask(task.toTaskRequest())
+            }.onError { error ->
+                if(error == DataError.Network.NO_INTERNET) {
+                    taskDao.insertModifiedTask(
+                        task.toModifiedTaskEntity(
+                            modificationType = ModificationType.Created
+                        )
+                    )
+                    return Result.Success(Unit)
+                }
+            }
```

Comment:
Could there not be more situations where the API call fails, and it would make sense to create this entry?
I think you plan to use this modified task to retry the network call later.